src/m365/spo/commands/tenant/tenant-site-membership-list.spec.ts

```diff
@@ -0,0 +1,196 @@
+      };
+
+      if (opts.url === `https://graph.microsoft.com/v1.0/sites/contoso.sharepoint.com:/sites/site?$select=id`) {
+        return { id: `contoso.sharepoint.com,${siteId},fb0a066f-c10f-4734-94d1-f896de4aa484` };
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, { options: { siteUrl: siteUrl } });
+    assert(loggerLogSpy.calledWith({
+      AssociatedOwnerGroup: membershipList,
+      AssociatedMemberGroup: membershipList,
+      AssociatedVisitorGroup: membershipList
+    }));
+  });
+
+  it('lists all site membership groups - just Owners group', async () => {
+    sinon.stub(request, 'get').callsFake(async (opts) => {
+      if (opts.url === `${adminUrl}/_api/SPO.Tenant/sites/GetSiteUserGroups?siteId='${siteId}'&userGroupIds=[0]`) {
+        return { value: [{ userGroup: membershipList }] };
+      };
+
+      if (opts.url === `https://graph.microsoft.com/v1.0/sites/contoso.sharepoint.com:/sites/site?$select=id`) {
+        return { id: `contoso.sharepoint.com,${siteId},fb0a066f-c10f-4734-94d1-f896de4aa484` };
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, { options: { siteUrl: siteUrl, role: "Owner" } });
+    assert(loggerLogSpy.calledWith({
+      AssociatedOwnerGroup: membershipList
+    }));
+  });
+
+  it('lists all site membership groups - just Members group', async () => {
+    sinon.stub(request, 'get').callsFake(async (opts) => {
+      if (opts.url === `${adminUrl}/_api/SPO.Tenant/sites/GetSiteUserGroups?siteId='${siteId}'&userGroupIds=[1]`) {
+        return { value: [{ userGroup: membershipList }] };
+      };
+
+      if (opts.url === `https://graph.microsoft.com/v1.0/sites/contoso.sharepoint.com:/sites/site?$select=id`) {
+        return { id: `contoso.sharepoint.com,${siteId},fb0a066f-c10f-4734-94d1-f896de4aa484` };
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, { options: { siteUrl: siteUrl, role: "Member" } });
+    assert(loggerLogSpy.calledWith({
+      AssociatedMemberGroup: membershipList
+    }));
+  });
+
+  it('lists all site membership groups - just Visitors group', async () => {
+    sinon.stub(request, 'get').callsFake(async (opts) => {
+      if (opts.url === `${adminUrl}/_api/SPO.Tenant/sites/GetSiteUserGroups?siteId='${siteId}'&userGroupIds=[2]`) {
+        return { value: [{ userGroup: membershipList }] };
+      };
+
+      if (opts.url === `https://graph.microsoft.com/v1.0/sites/contoso.sharepoint.com:/sites/site?$select=id`) {
+        return { id: `contoso.sharepoint.com,${siteId},fb0a066f-c10f-4734-94d1-f896de4aa484` };
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, { options: { siteUrl: siteUrl, role: "Visitor" } });
+    assert(loggerLogSpy.calledWith({
+      AssociatedVisitorGroup: membershipList
+    }));
+  });
+
+  it('correctly handles error when site is not found for specified site URL', async () => {
```

Comment:
Let's test for a real API error, e.g. the site doesn't exist.

Reply:
Done